### PR TITLE
[rocksdb_admin] switched overlapping keys to be per segment specified

### PR DESCRIPTION
**Pull request by cpd85**

### CODE REVIEW

File: rocksdb_admin/admin_handler.cpp [hunk 1]
```diff
@@ -57,6 +57,9 @@ DEFINE_string(shard_config_path, "",
 DEFINE_bool(rocksdb_allow_overlapping_keys, false,
             "Allow overlapping keys in sst bulk load");
 
+DEFINE_string(allow_overlapping_keys_segments, "",
+              "comma separated list of segments supporting overlapping keys");
```
Comment (newpoo) [inline rocksdb_admin/admin_handler.cpp]: Can we add "it takes priority over rocksdb_allow_overlapping_keys"

File: rocksdb_admin/admin_handler.cpp [hunk 2]
```diff
@@ -238,6 +241,8 @@ AdminHandler::AdminHandler(
   if (db_manager_ == nullptr) {
     db_manager_ = CreateDBBasedOnConfig(rocksdb_options_);
   }
+  folly::split(",", FLAGS_allow_overlapping_keys_segments,
```
Comment (newpoo) [inline rocksdb_admin/admin_handler.cpp]: Can we make allow_overlapping_keys_segments_  a set/unordered_set and use folly::splitTo()?
Though we don't really care about performance here, the code for looking up will be more elegant.

File: rocksdb_admin/admin_handler.cpp (hunk 1 above)
Comment (cpd85) [reply rocksdb_admin/admin_handler.cpp]: I'm not sure thats accurate. If FLAGS_rocksdb_allow_overlapping_keys = true and a segment doesn't exist in allow_overlapping_keys_segments, we still allow overlapping keys for the segment

File: rocksdb_admin/admin_handler.cpp [hunk 3]
```diff
@@ -806,7 +817,14 @@ void AdminHandler::async_tm_addS3SstFilesToDB(
   }
 
   clearMetaData(request->db_name);
-  if (!FLAGS_rocksdb_allow_overlapping_keys) {
+
+  bool allow_overlapping_keys = allow_overlapping_keys_segments_.find(
+                                    admin::DbNameToSegment(request->db_name)) !=
+                                allow_overlapping_keys_segments_.end();
```
Comment (newpoo) [inline rocksdb_admin/admin_handler.cpp]: fix indention.

**Pull request merged**